mwclient/client.py

```diff
@@ -979,7 +993,16 @@ def ask(self, query, title=None):
         while offset is not None:
             results = self.raw_api('ask', query='{query}|offset={offset}'.format(
                 query=query, offset=offset), http_method='GET', **kwargs)
-
-            offset = results.get('query-continue-offset')
-            for key, value in results['query']['results'].iteritems():
-                yield {key: value}
+            if results.get('error'):
```

Comment:
Since `self.handle_api_result(results)` will check for this, you don't need to do it here. You can replace this and the next three lines with just `self.handle_api_result(results)`